arch/arm/core/cortex_a_r/reset.S

```diff
@@ -56,6 +56,15 @@ SECTION_SUBSEC_FUNC(TEXT, _reset_section, __start)
     cmp r0, #MODE_HYP
     bne EL1_Reset_Handler
 
+#if defined(CONFIG_CPU_CORTEX_R52)
+#if (CONFIG_CPU_CORTEX_R52_ICACHE_FLASH_WAY > 0) || \
+    (CONFIG_CPU_CORTEX_R52_DCACHE_FLASH_WAY > 0)
+    ldr r0, =IMP_CSCTLR(CONFIG_CPU_CORTEX_R52_ICACHE_FLASH_WAY,
+                        CONFIG_CPU_CORTEX_R52_DCACHE_FLASH_WAY)
+    mcr p15, 1, r0, c9, c1, 0
```

Comment:
this should be done in inside the EL1_Reset_Handler because Zephyr may run on top of a hypervisor and access to IMP_CSCTLR can be trapped to EL2. Current implementation would only allow to configure cache segregation when Zephyr boots on EL2.

Reply:
I have updated, but not very clear. The cache segregation configuration is only permitted before the (EL2, EL1) caches have ever been enabled (after system reset). I'm not sure for the hypervisor case, i thought that the hypervisor always configures the cache segregation. Unless it does not enable EL2 cache before switch to EL1? or it replies on the trap to do that (i.e trap to EL2 --> EL2 configures cache segregation --> EL2 enables EL2 caches --> ...)

Reply:
that's a valid question but I don't have the answer. Arm docs mention that access to this register can be trapped to EL2 HCR.TIDCP bit is set, but I wouldn't know what the behavior of the hypervisor shall be in that case.